receiver/hostmetricsreceiver/hostmetrics_receiver.go

```diff
@@ -17,19 +17,27 @@ package hostmetricsreceiver
 import (
 	"context"
 	"fmt"
+	"time"
 
+	gopshost "github.com/shirou/gopsutil/host"
+	"go.opencensus.io/trace"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector/component"
 	"github.com/open-telemetry/opentelemetry-collector/component/componenterror"
 	"github.com/open-telemetry/opentelemetry-collector/consumer"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdata"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/pdatautil"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
 	"github.com/open-telemetry/opentelemetry-collector/receiver/hostmetricsreceiver/internal"
 )
 
 // Receiver is the type that scrapes various host metrics.
 type Receiver struct {
```

Comment:
Does this need to be exported? Consider to hide all structs and functions that don't need to be public.

Reply:
I changed the `Receiver` struct, all `Scraper` structs, and `New...()` functions to be non-public